src/app/pages/courses-page/pages/course-page/components/course-form/course-form.component.ts

```diff
@@ -46,31 +48,50 @@ export class CourseFormComponent implements OnInit {
 
   @Output()
   private cancelEvent = new EventEmitter<void>();
+  @Output()
+  private saveEvent = new EventEmitter<ICourse>();
 
   private searchStringSubject = new Subject<string>();
 
   constructor(private coursesService: CoursesService) { }
 
   public ngOnInit(): void {
-    this.createForm();
     this.initializeAuthors();
   }
 
+  ngOnChanges(): void {
+    this.createForm();
+  }
+
   public searchHints(searchString: string): void {
     this.searchStringSubject.next(searchString);
   }
 
   public cancel(): void {
     this.cancelEvent.emit();
   }
+  public save(): void {
+    const authorsNames = this.courseForm.value.authors as string[];
+
+    this.coursesService.getListOfAuthors()
+      .pipe(
+        map(authors => authors.filter(author => authorsNames.includes(author.name)))
+      ).subscribe(authors => {
+        const formData = { ...this.courseForm.value };
+        formData.authors = authors;
+        const course = formData as ICourse;
+        this.saveEvent.emit(course);
+      });
+
+  }
 
   public createForm(): void {
     this.courseForm = new CourseForm({
-      [CourseFormControl.Title]: this.course.title,
+      [CourseFormControl.Name]: this.course.name,
       [CourseFormControl.Description]: this.course.description,
       [CourseFormControl.Duration]: this.course.duration,
-      [CourseFormControl.Date]: this.course.creationDate,
-      [CourseFormControl.Authors]: this.course.authors,
+      [CourseFormControl.Date]: this.course.date,
+      [CourseFormControl.Authors]: this.course.authors.map(author => author.name),
     });
   }
 
```

Comment:
Вижу Лехин почерк в создании форм))) но у него это убусловлено тем что форма строится по приходящей с бэка метадате. А конкретно в твоем случае это усложнение потому как заранее исвестен состав формы. В общем то все можно и проще сделать через FormBuilder

Reply:
Нужно ли переделывать?

Reply:
Такой вариант тоже имеет право на существование, но в данной способ усложнен и избыточен. 
```suggestion
this.form = this.formBuilder.group({
  firstname: ['', [validator]],
  lastname: ['', [validator]],
  email: ['', [validator]],
  gender: ['', [validator]],
  isMarried: ['', [validator]],
  country: ['', [validator]],
});
```

Reply:
Вот смотри. Очень лаконично и наглядно.

Reply:
Поправь меня пожалуйста где я не так тебя понял. Ты предлагаешь:
- Избавиться от енама с именами для контролов
- Убрать класс CourseForm.
- Валидаторы перенести из класса CourseForm в текущий класс
- Использовать формбилдер для создания формы.
- Остальную логику оставить без изменений (если в компонент приходят данные курса, то создаем форму)